redash/authentication/google_oauth.py

```diff
@@ -113,7 +123,7 @@ def authorized():
 
     if not verify_profile(org, profile):
         logger.warning("User tried to login with unauthorized domain name: %s (org: %s)", profile['email'], org)
-        flash("Your Google Apps account ({}) isn't allowed.".format(profile['email']))
+        session['relogin'] = '1'
         return redirect(url_for('redash.login', org_slug=org.slug))
```

Comment:
Instead of relying on a session variable, let's pass a query string option to the `redash.login` route, which will tell it to ask the user to pick a new account (I guess this is what `extra['prompt'] = 'consent'` does).

Reply:
@alison985 What he means is that we can pass a query parameter to the redirected URL here (Flask will append any param that isn't matching a route path parameter as a query parameter instead), e.g.:

```python
# will turn into /user/login?prompt=consent
return redirect(url_for('redash.login', org_slug=org.slug, force='true'))
```

Then in the `handlers/authentication.py:login` function when it creates the `google_auth_url` variable using the `get_google_auth_url` function we can pass the `force` value along as a query parameter to the `google_oauth.authorize_org` or `google_oauth.authorize` view.

Then in the specific Google login view instead of checking the session check the query parmas:

```python
if 'force' in request.args:
    extra['prompt'] = 'consent'
```